### PR TITLE
[FIX] point_of_sale: block order validation if rpc error

### PR DESCRIPTION
**Pull request by caburj**
When validating an order, it's possible that a server error will be encountered.
With the assumption that the error shown to the cashier is actionable, we should
not proceed to the receipt screen (next screen), and should not mark the order
as paid so that the cashier will have a chance to rectify the issue.

### CODE REVIEW

**Comment by caburj** [comment]
Hi @odoo/rd-security team. We'd like to merge this fix but we're blocked by the code we introduce in the test. In the test, we want to observe a server error so in the test, we patch the main server method to raise an error.

File: addons/point_of_sale/tests/test_frontend.py
Comment (caburj) [reply addons/point_of_sale/tests/test_frontend.py:1225]: Thank you for the suggestion. However, if I applied the suggestion and remove the `assertLogs` context, I get the following log:

Screenshot 2024-04-25 at 10 56 38

I actually expect the following log:

Screenshot 2024-04-25 at 10 57 27

I added a commit to specify the type of log I'm expecting. I'll squash when you're okay with the changes.

Comment (caburj) [reply addons/point_of_sale/tests/test_frontend.py:1225]: > Get the class from the registry instead of deriving it from an instance.

I just saw that comment. I'm not sure if I understood it correctly, but after some debugging and doing `type(self.env['pos.order'])` is just giving the same result as `self.env.registry.models['pos.order']`. Do I need to replace my use of `type(self.env['pos.order'])`?

Comment (mart-e) [reply addons/point_of_sale/tests/test_frontend.py:1225]: > Do I need to replace my use of `type`?

Yes please

Comment (caburj) [reply addons/point_of_sale/tests/test_frontend.py]: I applied the following change to avoid the use of type and squash the commits.

Screenshot 2024-04-29 at 09 37 44

**Review by vlst-odoo — APPROVED** [review]
robodoo r+

**Pull request closed without merging**
